src/anndata/abc.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import ClassVar, Literal
+
+    import numpy as np
+    from scipy.sparse import csc_matrix, csr_matrix
+
+    from .compat import SpArray
+
+
+__all__ = ["CSRDataset", "CSCDataset"]
+
+
+class _AbstractCSDataset(ABC):
+    """Base for the public API for CSRDataset/CSCDataset."""
+
+    format: ClassVar[Literal["csr", "csc"]]
+    """The format of the sparse matrix."""
+
+    shape: tuple[int, int]
+    """Shape of the matrix."""
+
+    dtype: np.dtype
+    """The :class:`numpy.dtype` of the `data` attribute of the sparse matrix."""
+
+    backend: Literal["zarr", "hdf5"]
+    """Which file type is used on-disk."""
+
+    # TODO: index type
```

Comment:
Export that in `types`? I noticed it in a few spots in the github search for `_code`

Reply:
Do you mean `_types` and `_core`?

Reply:
Sorry! `typing`!

Reply:
So we should export `Index` in `typing`